### PR TITLE
#298 Replace LRAClientOps#isLRAFinished with afterLRA calls

### LINKED ISSUE #298: TCK: Replace LRAClientOps#isLRAFinished with afterLRA calls
As per Gitter discussion, replace the current way of checking of active LRA in the TCK (try enlisting) with direct SPI.

### PR DESCRIPTION
**Pull request by xstefank**
resolves #298

### CODE REVIEW

**Comment by xstefank** [comment]
@ochaloup can you also please take a look?

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/LRAClientOps.java [hunk 1]
```diff
@@ -246,10 +182,11 @@ private void cancelCancelation(URI lraId) {
     }
 
     void cleanUp(Logger logger, String testName) {
-        lraTasks.forEach((lra, future) -> {
+        for (Iterator<Map.Entry<LRATask, ScheduledFuture<?>>> it = lraTasks.entrySet().iterator(); it.hasNext();) {
```
[the remark this answers is not in the document]
Comment (xstefank) [reply tck/src/main/java/org/eclipse/microprofile/lra/tck/LRAClientOps.java:185]: this should be now fixed in https://github.com/eclipse/microprofile-lra/pull/324. Reverting.

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java [hunk 2]
```diff
@@ -105,4 +113,78 @@ public void waitForEndPhaseReplay(URI lraId) {
...
+        return getLRAEndStatus(lra, lraMetricType -> lraMetricService.getMetric(lraMetricType, lra, resourceName)) != null;
+    }
+
+    /**
+     * Returns whether the passed LRA is in finished state.
+     *
+     * @param lra the LRA to test
+     * @return whether or not an LRA has finished
+     */
+    public boolean isLRAFinished(URI lra) {
```
[the remark this answers is not in the document]
Comment (xstefank) [reply tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java:173]: I would leave this in test service as this is not intended for metrics.

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java [hunk 3]
```diff
@@ -105,4 +115,83 @@ public void waitForEndPhaseReplay(URI lraId) {
...
+    }
+
+    /**
+     * Returns the end {@link LRAStatus} of the passed LRA or null if the LRA is not in end state.
+     *
+     * @param lra the LRA whose end status is being queried
+     * @param metricFunction the metric function to be used to get the metrics
+     * @return the end status of the LRA (or null if the LRA has not yet finished)
+     */
+    private LRAStatus getLRAEndStatus(URI lra, Function<LRAMetricType, Integer> metricFunction) {
```
Comment (mmusgrov) [inline tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java:184]: This method is no longer used. Otherwise the PR looks good

**Review by mmusgrov — CHANGES_REQUESTED** [review]
There is just one comment about an unused method.

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java (hunk 3 above)
Comment (mmusgrov) [reply tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java:184]: @xstefank When you have resolved this comment we can look at getting this (and the remaining PRs merged). That will allow us to produce a release of milestone 1.

Comment (xstefank) [reply tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java:184]: sorry, PTO. On it.

**Review by mmusgrov — DISMISSED** [review]

**Comment by xstefank** [comment]
only rebase

**Review by mmusgrov — APPROVED** [review]
Changes still look good.

**Pull request merged**